scrapy/extensions/robotstxtparser.py

```diff
@@ -0,0 +1,52 @@
+from six.moves.urllib_robotparser import RobotFileParser
+
+from scrapy.utils.python import to_native_str
+
+class BaseRobotsTxtParser():
+    def __init__(self, robotstxt_url):
```

Comment:
Do parsers need to know the URL of the `robots.txt` file? Does it affect rules?

Reply:
No. We don't need to feed the URL of the `robots.txt` file since we are not dependent on them to fetch the file. The URL of `robots.txt` does not affect rules. I am changing this.

Reply:
I think, `robotstxt_url` is used as an argument in `RobotFileParser`'s constructor to properly catch domain name. We need to check `RobotFileParser` behaviour in case of domain mismatch between `robotstxt_url` and tested `url`.

Reply:
This doesn't seem to be the case. I looked at `RobotFileParser` source code. It uses `urllib.parse.urlparse` to separate domain name and path. It seems to be the case with both Python 2.7 and Python 3.X .The URL is only used to fetch `robots.txt` file.

```Python
>>> rp = RobotFileParser()
>>> rp.set_url('https://www.google.com/robots.txt')
>>> content = ("User-agent: abc \n"
... "Disallow: /disallowed \n"
... "Alloe: /allow \n")
>>> rp.parse(content.splitlines())
>>> rp.can_fetch('abc', 'https://www.google.com/disallowed')
False
>>> rp.can_fetch('abc', 'https://www.google.com/allowed')
True
>>> rp.can_fetch('abc', 'https://www.facebook.com/disallowed')
False
>>> rp.can_fetch('abc', 'https://www.facebook.com/allowed')
True
```

Reply:
In that case we should drop argument from constructor signature

Reply:
Great research!